src/checkers/gnomechecker.py

```diff
@@ -70,18 +70,21 @@ async def check(self, external_data: ExternalBase):
         if constraints:
             filtered_versions = filter_versions(filtered_versions, constraints)
 
-        if stable_only:
-            try:
-                latest_version = list(filter(_is_stable, filtered_versions))[-1]
-            except IndexError:
+        try:
+            if stable_only:
+                try:
+                    latest_version = list(filter(_is_stable, filtered_versions))[-1]
+                except IndexError:
+                    latest_version = filtered_versions[-1]
+                    log.warning(
+                        "Couldn't find any stable version for %s, selecting latest %s",
```

Comment:
What's the rationale for this (existing) behaviour?

Reply:
IIRC it had something to do with the `_is_stable` logic being unreliable in general.
I guess we can drop this warning and just error out.

Reply:
Ok, let's leave it as is for now.